pgdrive/scene_creator/blocks/bottleneck.py

```diff
@@ -0,0 +1,297 @@
+            inner_lane_line_type=LineType.BROKEN
+        ) and no_cross
+        no_cross = CreateAdverseRoad(
+            socket_road,
+            self.block_network,
+            self._global_network,
+            inner_lane_line_type=LineType.BROKEN,
+            side_lane_line_type=LineType.SIDE,
+            center_line_type=LineType.CONTINUOUS
+        ) and no_cross
+
+        negative_sockect_road = -socket_road
+        self.add_sockets(BlockSocket(socket_road, negative_sockect_road))
+
+        # part 2, circular part
+        for index, lane in enumerate(self.positive_lanes[straight_lane_num:], 1):
+            lateral_dist = index * self.lane_width / 2
+            inner_node = self.road_node(1, index)
+            side_line_type = LineType.SIDE if index == self.positive_lane_num - straight_lane_num else LineType.NONE
+
+            # positive part
+            circular_1, circular_2, _ = create_wave_lanes(lane, lateral_dist, self.BOTTLENECK_LEN, 5, self.lane_width)
+            road_1 = Road(start_ndoe, inner_node)
+            no_cross = CreateRoadFrom(
+                circular_1,
+                1,
+                road_1,
+                self.block_network,
+                self._global_network,
+                center_line_type=LineType.NONE,
+                side_lane_line_type=side_line_type,
+                inner_lane_line_type=LineType.NONE
+            ) and no_cross
+            road_2 = Road(inner_node, self.road_node(0, 0))
+            no_cross = CreateRoadFrom(
+                circular_2,
+                1,
+                road_2,
+                self.block_network,
+                self._global_network,
+                center_line_type=LineType.NONE,
+                side_lane_line_type=side_line_type,
+                inner_lane_line_type=LineType.NONE
+            ) and no_cross
+
+            # adverse part
+            lane = negative_sockect_road.get_lanes(self.block_network)[-1]
+            circular_2, circular_1, _ = create_wave_lanes(
+                lane, lateral_dist, self.BOTTLENECK_LEN, 5, self.lane_width, False
+            )
+            road_2 = -road_2
+            no_cross = CreateRoadFrom(
+                circular_2,
+                1,
+                road_2,
+                self.block_network,
+                self._global_network,
+                center_line_type=LineType.NONE,
+                side_lane_line_type=side_line_type,
+                inner_lane_line_type=LineType.NONE
+            ) and no_cross
+
+            road_1 = -road_1
+            no_cross = CreateRoadFrom(
+                circular_1,
+                1,
+                road_1,
+                self.block_network,
+                self._global_network,
+                center_line_type=LineType.NONE,
+                side_lane_line_type=side_line_type,
+                inner_lane_line_type=LineType.NONE
+            ) and no_cross
+
+        return no_cross
+
+
+class OutBottleneck(Bottleneck):
```

Comment:
"Split"

Reply:
Ok, names will be changed to Merge and Split